packages/core/src/odata/common/entity-deserializer.ts

```diff
@@ -202,7 +205,12 @@ export function entityDeserializer(
     return complexType._propertyMetadata
       .map(property => ({
         ...(typeof json[property.originalName] !== 'undefined' && {
-          [property.name]: isComplexTypeNameSpace(property.type)
+          [property.name]: property.isCollection
```

Comment:
I find this kind of chaining of tenary operators in combination with the spread very hard to read. I would prefer a function(json,originalName){
-undfined check->return,
-isCollection->return deserializeCollectionType(),
-isComplect->return deserializeComplexType().
-isEdmToTs->return edmToTs
Throw exception because no case is found which should not be right?

Reply:
I extracted this to a separate function